client/src/features/action/assets/Assets.tsx

```diff
@@ -78,9 +79,24 @@ function Assets(): JSX.Element {
     });
   };
 
+  const [selectedAssetType, setSelectedAssetType] = React.useState<'donation' | 'request'>(
+    'donation',
+  );
+  const [donations, setDonations] = React.useState<Asset[]>([]);
+  const [needs, setNeeds] = React.useState<Asset[]>([]);
+
   React.useEffect(() => {
     // fetch assets with querySearchText
-  }, [location]);
+    fetch(`http://localhost:3001/api/assets?type=${selectedAssetType}&title=${querySearchText}`)
+      .then((resp) => resp.json())
```

Comment:
looks like filtering on `title` is blocking this request
why don't we update the url to `fetch('blah?type={}${querySearchText ? '&title=${querySearchText}' : ''`}'`

Reply:
I made the change, but I not sure why. Is there somewhere on the backend I should look to understand what happening there?

Reply:
yep, everything after the `?` in the url is the query params. so `title` finds its way into `@Query()` and the `getAssetsDto` which is what filters the assets list
happy to discuss that further if needed